ga4gh/datamodel/sequenceAnnotations.py

```diff
@@ -0,0 +1,415 @@
+        strands = [protocol.Strand.POS_STRAND, protocol.Strand.NEG_STRAND]
+        feature.strand = randomNumberGenerator.choice(strands)
+        feature.attributes = protocol.Attributes()
+        feature.attributes.vals = {"gene_name": ["Frances", ],
+                                   "gene_type": ["mRNA", ],
+                                   "gene_status": ["UNKNOWN", ]}
+        return feature
+
+    def getFeature(self, compoundId):
+        """
+        Fetches a simulated feature by ID.
+        :param compoundId: any non-null string
+        :return: A simulated feature with id set to the same value as the
+            passed-in compoundId.
+        ":raises: exceptions.ObjectWithIdNotFoundException if None is passed
+            in for the compoundId.
+        """
+        if compoundId is None:
+            raise exceptions.ObjectWithIdNotFoundException(compoundId)
+        randomNumberGenerator = random.Random()
+        randomNumberGenerator.seed(self._randomSeed)
+        feature = self._generateSimulatedFeature(randomNumberGenerator)
+        feature.id = str(compoundId)
+        feature.parentId = ""  # TODO: Test with nonempty parentIDs?
+        return feature
+
+    def getFeatures(self, referenceName, start, end,
+                    pageToken, pageSize,
+                    featureTypes=[], parentId=None):
+        """
+        Returns a set number of simulated features.
+        :param referenceName: name of reference to "search" on
+        :param start: start coordinate of query
+        :param end: end coordinate of query
+        :param pageToken: None or int
+        :param pageSize: None or int
+        :param ontologyTerms: optional list of ontology terms to limit query
+        :param parentId: optional parentId to limit query.
+        :return: Yields feature, nextPageToken pairs.
+            nextPageToken is None if last feature was yielded.
+
+        """
+        randomNumberGenerator = random.Random()
+        randomNumberGenerator.seed(self._randomSeed)
+        numFeatures = 10
+        if pageToken is not None:
+            nextPageToken = int(pageToken)
+        else:
+            nextPageToken = 0
+        for featureId in ["feature{}".format(x) for x in xrange(numFeatures)]:
+            gaFeature = self._generateSimulatedFeature(randomNumberGenerator)
+            gaFeature.id = self.getCompoundIdForFeatureId(featureId)
+            match = (gaFeature.start < end and
+                     gaFeature.end > start and
+                     gaFeature.referenceName == referenceName and
+                     (gaFeature.featureType in featureTypes or
+                      len(featureTypes) == 0))
+            if match:
+                gaFeature.parentId = ""  # TODO: Test with nonempty parentIDs?
+                if nextPageToken < numFeatures - 1:
+                    nextPageToken += 1
+                else:
+                    nextPageToken = None
+                yield gaFeature, \
```

Comment:
The logic here is a bit touchy, and took me a couple of tries to get it right. I appreciate the suggestion but want to leave it as is for now.